components/core/src/package/install.rs

```diff
@@ -251,6 +252,62 @@ impl PackageInstall {
         self.read_deps(MetaFile::TDeps)
     }
 
+    /// Returns a Rust representation of the mappings defined by the `pkg_env` plan variable.
+    ///
+    /// # Failures
+    ///
+    /// * The package contains a Environment metafile but it could not be read or it was malformed.
+    pub fn environment(&self) -> Result<HashMap<String, String>> {
```

Comment:
The logic is very similiar to https://github.com/habitat-sh/habitat/pull/1830/files#diff-b9b27ac18067cb62ebad852e9abc0b10R288 except the object is for EnvironmentSep vs Environment

This could be DRY.